syntax = "proto3";

service Power {

    rpc PowerOn (PowerRequest) returns (stream StatusResponse) {
    }

    rpc PowerOff (PowerRequest) returns (stream StatusResponse) {
    }

    rpc HardReset (PowerRequest) returns (stream StatusResponse) {
    }
}

message PowerRequest {
    bytes token = 1;
    int32 delay_sec = 2;
}

message StatusResponse {
    string info = 1;
}

service GPIO {
    rpc Signal (SignalRequest) returns (SignalResponse) {
    }
}

message SignalRequest {
    enum Signal {
        POWERON = 0;
        POWEROFF = 1;
        HARDRESET = 2;
    }
    Signal signal = 1;
}

message SignalResponse {
    bool info = 1;
}
